projects/addon-charts/components/arc-chart/arc-chart.component.ts

```diff
@@ -94,13 +105,10 @@ export class TuiArcChartComponent {
 
     constructor(
         @Inject(DomSanitizer) private readonly sanitizer: DomSanitizer,
-        @Inject(ChangeDetectorRef) cdr: ChangeDetectorRef,
+        @Inject(ChangeDetectorRef) private readonly cd: ChangeDetectorRef,
+        @Self() @Inject(TuiDestroyService) private readonly destroy$: Observable<void>,
     ) {
-        // So initial animation works
-        setTimeout(() => {
-            this.initialized = true;
-            cdr.markForCheck();
-        });
+        this.initializeAnimation();
```

Comment:
If we do not move it to a method we can keep `destroy$` and `cdr` from storing as class props. Also let's use `tuiWatch(cdr)`

Reply:
fixed